src/Mafia/Process.hs

```diff
@@ -174,6 +182,37 @@ renderProcessError = \case
 
 ------------------------------------------------------------------------
 
+createProcess :: MonadIO m => Process.CreateProcess -> m (Maybe Handle, Maybe Handle, Maybe Handle, Process.ProcessHandle)
+createProcess = liftIO . Process.createProcess
+
+-- Spawn a new process, and if we get a ctrl-c, make absolutely sure everything we started is finished.
+createProcessAnnihilate :: MonadIO m => Process.CreateProcess -> m (Maybe Handle, Maybe Handle, Maybe Handle, Process.ProcessHandle)
+createProcessAnnihilate cp = do
+  (a, b, c, pid) <- createProcess cp { Process.create_group = True }
+
+  pgid <- liftIO $ ProcessInternals.withProcessHandle pid getPgid
+  case pgid of
+   Just pgid' -> do
+    _ <- liftIO $ Signals.installHandler Signals.keyboardSignal (Signals.Catch $ killer pgid') Nothing
+    return ()
+   Nothing -> return ()
+
+  return (a, b, c, pid)
+ where
+  getPid = \case
+   ProcessInternals.OpenHandle i -> Just i
+   ProcessInternals.ClosedHandle _ -> Nothing
+
+  getPgid i = case getPid i of
+    Nothing -> return Nothing
+    Just h -> do
+      let ignoreIOE (_ :: IOException) = return Nothing
+      handle ignoreIOE (Just <$> Posix.getProcessGroupIDOf h)
+
+  -- SIGQUIT
+  killer = Signals.signalProcessGroup Signals.keyboardTermination
```

Comment:
Any reason `getPid`, `getPgid` and `killer` shouldn't just be top level functions? And in that case maybe `killer` should be called `sigquit` or something to that effect?